README.md

```diff
@@ -21,128 +19,504 @@ composer require php-llm/llm-chain
-* **YouTube Transcriber** Tool to fetch transcriptions from YouTube videos.
+Those models are provided by different **platforms**, like OpenAI, Azure, Replicate, and others.
 
-Usage Examples
---------------
+#### Example Instantiation
 
-See [examples](examples) to run example implementations using this library.
-Depending on the example you need to export different environment variables
-for API keys or deployment configurations or create a `.env.local` based on `.env` file.
+```php
+use PhpLlm\LlmChain\OpenAI\Model\Embeddings;
+use PhpLlm\LlmChain\OpenAI\Model\Gpt;
+use PhpLlm\LlmChain\OpenAI\Model\Gpt\Version;
+use PhpLlm\LlmChain\OpenAI\Platform\OpenAI;
+use Symfony\Component\HttpClient\HttpClient;
+
+// Platform: OpenAI
+$platform = new OpenAI(HttpClient::create(), $_ENV['OPENAI_API_KEY']);
+
+// Language Model: GPT (OpenAI)
+$llm = new Gpt($platform, Version::gpt4oMini()); 
+
+// Embeddings Model: Embeddings (OpenAI)
+$embeddings = new Embeddings($platform);
+```
+
+#### Supported Models & Platforms
+
+* Language Models
+  * [OpenAI's GPT](https://platform.openai.com/docs/models/overview) with [OpenAI](https://platform.openai.com/docs/overview) and [Azure](https://learn.microsoft.com/azure/ai-services/openai/concepts/models) as Platform
+  * [Anthropic's Claude](https://www.anthropic.com/claude) with [Anthropic](https://www.anthropic.com/) as Platform
+* Embeddings Models
+  * [OpenAI's Text Embeddings](https://platform.openai.com/docs/guides/embeddings/embedding-models) with [OpenAI](https://platform.openai.com/docs/overview) and [Azure](https://learn.microsoft.com/azure/ai-services/openai/concepts/models) as Platform
+  * [Voyage's Embeddings](https://docs.voyageai.com/docs/embeddings) with [Voyage](https://www.voyageai.com/) as Platform
+
+See [issue #28](https://github.com/php-llm/llm-chain/issues/28) for planned support of other models and platforms.
+
+### Chain & Messages
+
+The core feature of LLM Chain is to interact with language models via messages. This interaction is done by sending
+a **MessageBag** to a **Chain**, which takes care of LLM invokation and response handling.
+
+Messages can be of different types, most importantly `UserMessage`, `SystemMessage`, or `AssistantMessage`, and can also
+have different content types, like `Text` or `Image`.
+
+#### Example Chain call with messages
+
+```php
+use PhpLlm\LlmChain\Chain;
+use PhpLlm\LlmChain\Message\MessageBag;
+use PhpLlm\LlmChain\Message\SystemMessage;
+use PhpLlm\LlmChain\Message\UserMessage;
+
+// LLM instantiation
+
+$chain = new Chain($llm);
+$messages = new MessageBag(
+    new SystemMessage('You are a helpful chatbot answering questions about LLM Chain.'),
+    new UserMessage('Hello, how are you?'),
+);
+$response = $chain->call($messages);
+
+echo $response->getContent(); // "I'm fine, thank you. How can I help you today?"
+```
+
+The `MessageInterface` and `Conent` interface help to customize this process if needed, e.g. additional state handling.
```

Comment:
```suggestion
The `MessageInterface` and `Content` interface help to customize this process if needed, e.g. additional state handling.
```

Reply:
Maybe rename `Content` to `ContentInterface`?

Reply:
It was like that before and usually i try to avoid the Interface prefix - but we're not consistent here right now

Reply:
Fair enough, i think its a bit more readable with the Interface suffix. but anyway there is i think still a typo here `Conent` => `Content`

Reply:
Yea, good catch, merged that one